File: BOJ_java11/done/Boj_1021.java
package done;

import java.util.*;

public class Boj_1021 {

	static int calc(Deque<Integer> dq, int target) {
		int order = 0, size = dq.size();
		int num;
		while ((num = dq.removeFirst()) != target) {
			order++;
			dq.add(num);
		}
		order = Math.min(order, size - order);
		return order;
	}

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int N, M;
		Deque<Integer> dq = new LinkedList<>();

		N = sc.nextInt();
		M = sc.nextInt();
		int sum = 0;

		for (int i = 1; i <= N; i++)
			dq.add(i);

		for (int i = 0; i < M; i++)
			sum += calc(dq, sc.nextInt());

		sc.close();

		System.out.print(sum);

	}

}

File: _boj solution/Boj1021.py

from collections import deque

def calc(dq:deque,  target:int):
    order =0
    size = len(dq)
    while dq[0] != target:
        order+=1
        dq.append(dq.popleft())

    dq.popleft()
    
    return min(order, size-order)

def main():
    N,M = map(int, input().split())
    targets = list (map(int, input().split()))

    dq = deque([i for i in range(1,N+1)])

    sum=0
    for target in targets:
        sum += calc(dq, target)

    print(sum)

    
main()


    
